### ISSUE TITLE
[windows] Terminal background colour

### ISSUE THREAD

**Issue by gvanem** [google/benchmark#1487]
Hello folks, my first issue.

The function `ColorPrintf()` ass-u-mes that all users of this library have terminals with a black background colour.
I.e. using it in [Microsoft's STL](https://github.com/microsoft/STL.git) library and the `benchmark-std_copy.exe` program, the output like like this:

![STL-1](https://user-images.githubusercontent.com/945271/189484264-1b7ae7b6-8003-4ac8-ae32-0010261bd4c3.png)

Not very nice IMHO.

With my blue terminal background I'd like it to look like:
![STL-2](https://user-images.githubusercontent.com/945271/189484329-e720fd3b-138c-4b1f-a3e6-c212b473b5ab.png)

which I fixed with:
```diff
--- a/src/colorprint.cc  2022-09-10 14:58:12
+++ b/src/colorprint.cc 2022-09-10 14:54:32
@@ -136,13 +136,16 @@
   CONSOLE_SCREEN_BUFFER_INFO buffer_info;
   GetConsoleScreenBufferInfo(stdout_handle, &buffer_info);
   const WORD old_color_attrs = buffer_info.wAttributes;
+  WORD  new_attr;

   // We need to flush the stream buffers into the console before each
   // SetConsoleTextAttribute call lest it affect the text that is already
   // printed but has not yet reached the console.
   fflush(stdout);
+  new_attr = (buffer_info.wAttributes & ~7);
+  new_attr &= ~8;    // Since 'wAttributes' could have been hi-intensity at startup.
   SetConsoleTextAttribute(stdout_handle,
-                          GetPlatformColorCode(color) | FOREGROUND_INTENSITY);
+                          new_attr | GetPlatformColorCode(color) | FOREGROUND_INTENSITY);
   vprintf(fmt, args);

   fflush(stdout);
```

PS. I use [JPsoft's](https://jpsoft.com) 4NT as the shell.

**System**
  - Win-10 
  - Compiler and version: MSVC ver. 19.34.31721 for x64

**Comment by dominichamon** [comment]
could you create a pull request for this?

**Comment by gvanem** [comment]
I'm not very familiar with git pull request. But my attempt: https://github.com/google/benchmark/pull/1488

**Comment by imrichardcole** [comment]
Taking a look

**Comment by imrichardcole** [comment]
To confirm why this is happening (PR incoming) the issue can be captured like so:

![image](https://github.com/user-attachments/assets/feffd94b-a39d-4dee-b611-30630e4b4959)

In a blue background console, the 5th bit is lost at the point the green message is written.

**Comment by dmah42** [comment]
yeah i think we just assume the background colour is black and reset to that instead of reading the current bg colour.

**Comment by gvanem** [comment]
> yeah i think we just assume the background colour is black and reset to that instead of reading the current bg colour.

That's just what I wrote 2 years ago. So much for being a *Member of Google*.

**Comment by dmah42** [comment]
fixed by #1865
